src/tests/integration_tests/test_number_citation_validator.py

```diff
@@ -0,0 +1,186 @@
+import pytest
+from loguru import logger
+
+from sherpa_ai.agents import QAAgent
+from sherpa_ai.events import EventType
+from sherpa_ai.memory import SharedMemory
+from sherpa_ai.models.sherpa_base_chat_model import SherpaChatOpenAI
+from sherpa_ai.test_utils.llms import get_llm
+from sherpa_ai.tools import SearchTool
+import sherpa_ai.config as cfg
+from unittest.mock import patch
+from sherpa_ai.utils import extract_numbers_from_text
+
+
+
+@pytest.mark.parametrize(
+    "objective , input_data, expected_numbers",
+    [
+        # (
```

Comment:
Are these commented inputs active? If not, they should be removed. I think they make the test case very hard to understand.

Reply:
they are active but running them all at the same time will cause them to fail @20001LastOrder . so if you have a suggestion a way to keep them since they hold different aspects of the test that would be nice.

Reply:
Hmm, this is quite strange because each test cases are meant to be isolated. I'll try to allocate some time to see if I can figure out why this happens